### DISCUSSION TITLE
ImGui Windows [Lua]

### DISCUSSION THREAD

**Discussion by xesdoog** [YimMenu/YimMenu#3037] [Q&A]
Hello,

I have a question regarding ImGui windows drawn from Lua scripts.

This is an example code for a debug window from a script I'm currently working on. I'm tying the window drawing to a bool that turns true if you click on the script version 5 times:

```lua
if dbgWindow then
    ImGui.SetNextWindowSizeConstraints(450, 100, 600, 800)
    ImGui.Begin("##bsDebug", ImGuiWindowFlags.AlwaysAutoResize | ImGuiWindowFlags.NoTitleBar)
    ImGui.Text("Billionaire Services Debug");ImGui.SameLine(380)
    if ImGui.Button("Close") then
      dbgWindow = false
      dbgclc = 0
    end

-- some debug code goes here

    ImGui.End()
end
```

The window opens and closes fine but only when the script UI is open. Is there a way to keep the window drawn even when the menu or the script UI or both are closed? Like YimMenu's vehicle controller?

**Comment by pierrelasse** [comment]
You could technically add the `add_imgui` to every tab in the menu I think.

**Comment by xesdoog** [comment]
Coming back to this discussion because #3563 made it possible if anyone is still wondering if it's doable.
Thanks @tupoy-ya  ♥
